[BE] [Inductor] Re-Land Support TMA before strict 3.4 cutoff

njriasan opened the pull request:
Summary: Inductor's 3.4 Triton release is the most common used variant of Triton, but if someone is working with an alternative version of Triton this may not match. This moves the version check from 3.4 Triton to any variant that has support for the TMA APIs.

Test Plan:
Testing the previously failing test `inductor/test_torchinductor_strided_blocks.py::TritonTensorDescriptorTestCUDA::test_welford_non_block_pointer_cuda`

Rollback Plan:

Differential Revision: D80348643




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

njriasan commented:
@pytorchbot merge

malfet commented:
@pytorchbot revert -m "Looks like this breaks rocm, see https://hud.pytorch.org/hud/pytorch/pytorch/main/1?per_page=50&name_filter=rocm%20%2F%20linux-jammy-rocm-py3.10" -c nosignal

njriasan commented:
> https://hud.pytorch.org/hud/pytorch/pytorch/main/1?per_page=50&name_filter=rocm%20%2F%20linux-jammy-rocm-py3.10

Thanks! I'll rerun this test on rocm. I expect this PR shouldn't impact rocm because TMA isn't supported on AMD, but that probably points to an existing bug.

njriasan commented:
Doing a deeper dive it looks like this test is just broken. In particular I believe this check is setting `expected_num_block_pointers=6` because of TMA, which only works for Nvidia and not AMD. I'll update this test to work better.

njriasan committed:
[BE] [Inductor] Re-Land Support TMA before strict 3.4 cutoff (#160747)

Summary:

Inductor's 3.4 Triton release is the most common used variant of Triton, but if someone is working with an alternative version of Triton this may not match. This moves the version check from 3.4 Triton to any variant that has support for the TMA APIs.

Test Plan:
Testing the previously failing test `inductor/test_torchinductor_strided_blocks.py::TritonTensorDescriptorTestCUDA::test_welford_non_block_pointer_cuda`

Rollback Plan:

Differential Revision: D80348643

njriasan commented:
@pytorchmergebot merge

njriasan commented:
I see rocm failures on CI, but they seem unrelated and I'm not convinced these tests should be running. Before reverting this PR can we try fixing the CI issues in a followup PR. What I saw with the last failure is that it was broken on main but it just didn't run for some reason.

njriasan commented:
I added a fix here: https://github.com/pytorch/pytorch/pull/160974.

jeffdaily reviewed (commented):
Please fix in follow-up PR or I will revert this.  Breaking ROCm CI with this typo.

jeffdaily commented on test/inductor/test_torchinductor_strided_blocks.py:
```diff
@@ -1324,7 +1324,11 @@ class TritonBlockPointerTestGPU(BlockDescriptorTestBase):
+    not (
+        HAS_CUDA_AND_TRITON
+        and torch.cuda.get_device_capability()[0] >= 9
+        and torch.hip.version is None
```
Fixed in https://github.com/pytorch/pytorch/pull/160981.
